ospf6d/ospf6_asbr.c

```diff
@@ -2316,6 +2316,31 @@ void ospf6_asbr_redistribute_reset(struct ospf6 *ospf6)
 	}
 }
 
+void ospf6_asbr_redistribute_reconfigure(struct ospf6 *ospf6)
+{
+	int type;
+	struct ospf6_redist *red;
+
+	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
+		if (type == ZEBRA_ROUTE_OSPF6)
+			continue;
+		red = ospf6_redist_lookup(ospf6, type, 0);
+		if (!red)
+			continue;
+
+		if (type == DEFAULT_ROUTE) {
+			ospf6_redistribute_default_set(
+				ospf6, ospf6->default_originate);
+			continue;
+		}
+
+		ospf6_asbr_redistribute_unset(ospf6, red, type);
+		ospf6_redist_del(ospf6, red, type);
```

Comment:
The route-map configuration will get removed due to ospf6_asbr_redistribute_unset here. you can try testing "redistribute static route-map abc" and then change the router-id, the route-map config will get deleted. Another way is to loop through the external table and then originate the LSAs, I guess ospfv2 does that way.